backend/models/sequence.js

```diff
@@ -116,24 +224,36 @@ Sequence.getSequenceState = async (account, model, stateId) => {
 };
 
 Sequence.getList = async (account, model, branch, revision, cleanResponse = false) => {
-	const history = await History.getHistory(account, model, branch, revision);
+	let submodelBranch;
+	let sequencesQuery = {};
+
+	if (branch || revision) {
+		const history = await History.getHistory(account, model, branch, revision);
+
+		if (!history) {
+			throw responseCodes.INVALID_TAG_NAME;
+		}
 
-	if(!history) {
-		throw responseCodes.INVALID_TAG_NAME;
+		submodelBranch = "master";
+		sequencesQuery = {"$or":[{"rev_id": history._id}, {"rev_id": {"$exists": false}}]};
 	}
 
-	const refNodes = await getRefNodes(account, model, branch, revision, {project:1});
+	const refNodesBranch = revision ? undefined : "master";
+	const refNodes = await getRefNodes(account, model, refNodesBranch, revision, {project:1});
 	const submodels = refNodes.map(r => r.project);
 
-	const submodelSequencesPromises = Promise.all(submodels.map((submodel) => Sequence.getList(account, submodel, "master", null, cleanResponse)));
+	const submodelSequencesPromises = Promise.all(submodels.map((submodel) => Sequence.getList(account, submodel, submodelBranch, undefined, cleanResponse)));
+
+	const sequences = await db.find(account, sequenceCol(model), sequencesQuery);
 
-	const sequences = await db.find(account, sequenceCol(model), {"rev_id": history._id});
 	sequences.forEach((sequence) => {
-		sequence.teamspace = account;
-		sequence.model = model;
+		sequence.minDate = new Date((sequence.frames[0] || {}).dateTime).getTime();
+		sequence.maxDate = new Date((sequence.frames[sequence.frames.length - 1] || {}).dateTime).getTime();
+
+		delete sequence.frames;
```

Comment:
do a exclusion (i.e. `frames: 0`) projection instead doing this.

no point dragging the data out of the database just to ditch it.